src/main/java/jnr/ffi/Struct.java

```diff
@@ -101,13 +103,12 @@ protected final int addField(int sizeBits, int alignBits, Offset offset) {
         }
 
         protected final int addField(int sizeBits, int alignBits) {
-            final int off = resetIndex ? 0 : align(this.size, alignBits >> 3);
-            this.size = Math.max(this.size, off + (sizeBits >> 3));
-            this.minAlign = Math.max(this.minAlign, alignBits >> 3);
-            return off;
+            final int alignment = this.alignment.intValue() > 0 ? Math.min(this.alignment.intValue(), (alignBits >> 3)) : (alignBits >> 3);
```

Comment:
Try a Test when you put two unsigned8 field in a struct with padding set to 16. The first must have 0 offset, the second must have 16.